x-pack/plugins/infra/public/components/logging/log_entry_flyout/log_entry_actions_menu.tsx

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { EuiButtonEmpty, EuiContextMenuItem, EuiContextMenuPanel, EuiPopover } from '@elastic/eui';
+import { FormattedMessage } from '@kbn/i18n/react';
+import React, { useCallback, useMemo, useState } from 'react';
+import url from 'url';
+
+import chrome from 'ui/chrome';
+import { InfraLogItem } from '../../../graphql/types';
+
+const UPTIME_FIELDS = ['container.id', 'host.ip', 'kubernetes.pod.uid'];
+
+export const LogEntryActionsMenu: React.FunctionComponent<{
+  logItem: InfraLogItem;
+}> = ({ logItem }) => {
+  const [hide, isVisible, show] = useVisibility();
```

Comment:
I chose this style because it is self-documenting and can avoid unnecessary re-renders at the same time. It also avoids the boolean argument code smell.

I just noticed, though, that I should use an object instead of array as the return data structure to avoid confusion. A 3-tuple requires too much implicit knowledge for my taste.

Reply:
I'd be fine just handling it inline too, but I find it hard to formulate a suitable criterion to prefer one over the other.

Reply:
I'm good either way right now, and agree the object would be clearer.